### ISSUE TITLE
Integration tests stuck on iOS

### ISSUE THREAD

**Issue by AndreHaueisen** [flutter/flutter#112174]
## Steps to Reproduce

This issue has been reported before but it was closed. 
https://github.com/flutter/flutter/issues/99246

I anyone able to run integration tests on iOS?

```

[✓] Flutter (Channel stable, 3.3.0, on macOS 12.6 21G115 darwin-arm, locale en-BR)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc1)
[✓] Xcode - develop for iOS and macOS (Xcode 14.0)
[✓] Chrome - develop for the web
[✓] Android Studio (version 2021.3)
[✓] IntelliJ IDEA Community Edition (version 2022.2.2)
[✓] Connected device (3 available)
[✓] HTTP Host Availability

• No issues found!
andrehaueisen@Andres-Mac-mini app_coach_operations % flutter doctor -v
[✓] Flutter (Channel stable, 3.3.0, on macOS 12.6 21G115 darwin-arm, locale en-BR)
    • Flutter version 3.3.0 on channel stable at /Users/andrehaueisen/Development/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision ffccd96b62 (3 weeks ago), 2022-08-29 17:28:57 -0700
    • Engine revision 5e9e0e0aa8
    • Dart version 2.18.0
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.0-rc1)
    • Android SDK at /Users/andrehaueisen/Library/Android/sdk
    • Platform android-33, build-tools 33.0.0-rc1
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A309
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.3)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.13+0-b1751.21-8125866)

[✓] IntelliJ IDEA Community Edition (version 2022.2.2)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 70.0.5
    • Dart plugin version 222.4167.21

[✓] Connected device (3 available)
    • iPhone 14 Pro Max (mobile) • 83513B47-DABC-492B-A95D-6DA335D6553B • ios            • com.apple.CoreSimulator.SimRuntime.iOS-16-0
      (simulator)
    • macOS (desktop)            • macos                                • darwin-arm64   • macOS 12.6 21G115 darwin-arm
    • Chrome (web)               • chrome                               • web-javascript • Google Chrome 105.0.5195.125

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```

</details>

## Videos ##

[You can see the difference in behavior running on Android and iOS here](https://drive.google.com/drive/folders/1BGcZrSk4-Oq2ZEvKz3m0y5EvurP7-VSO?usp=sharing)

**Comment by huycozy** [comment]
Hi @AndreHaueisen 
I can run the test with the sample code in issue #99246.

Device info: `iPhone 14 (iOS 16.0)`
Xcode version: `14.0`

<details>
<summary>Logs</summary>

```bash
/Users/huynq/Documents/GitHub/flutter/bin/flutter --no-color test --machine --start-paused -d 9CF7DFF6-441D-4224-8C25-FFB35EF84E9C integration_test/issue_99246_ios_testt.dart
Testing started at 13:53 ...
Running pod install...                                           2,818ms
Running Xcode build...                                          
Xcode build done.                                           86.6s

══╡ EXCEPTION CAUGHT BY FLUTTER TEST FRAMEWORK ╞════════════════════════════════════════════════════
The following TestFailure was thrown running a test:
Expected: <2>
  Actual: <1>

When the exception was thrown, this was the stack:
#4      main.<anonymous closure>.<anonymous closure> (file:///Users/huynq/Documents/NEVERCODE.IO/RepoduceProjects/reproduce_issues/integration_test/issue_99246_ios_testt.dart:9:7)
#5      testWidgets.<anonymous closure>.<anonymous closure> (package:flutter_test/src/widget_tester.dart:171:29)
<asynchronous suspension>
<asynchronous suspension>
(elided one frame from package:stack_trace)

This was caught by the test expectation on the following line:
  file:///Users/huynq/Documents/NEVERCODE.IO/RepoduceProjects/reproduce_issues/integration_test/issue_99246_ios_testt.dart line 9
The test description was:
  Failing test
════════════════════════════════════════════════════════════════════════════════════════════════════

Test failed. See exception logs above.
The test description was: Failing test
```
</details>

<details>
<summary>Sample code</summary>

```dart
import 'package:flutter_test/flutter_test.dart';
import 'package:integration_test/integration_test.dart';

void main() {
  IntegrationTestWidgetsFlutterBinding.ensureInitialized();

  group('end-to-end test', () {
    testWidgets('Failing test', (WidgetTester tester) async {
      expect(1, 2);
    });
  });
}
```
</details>

<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel stable, 3.3.2, on macOS 12.6 21G115 darwin-x64, locale en-VN)
    • Flutter version 3.3.2 on channel stable at /Users/huynq/Documents/GitHub/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision e3c29ec00c (6 days ago), 2022-09-14 08:46:55 -0500
    • Engine revision a4ff2c53d8
    • Dart version 2.18.1
    • DevTools version 2.15.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 13.3)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 13E113
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.71.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.48.0

[✓] Connected device (2 available)
    • macOS (desktop) • macos  • darwin-x64     • macOS 12.6 21G115 darwin-x64
    • Chrome (web)    • chrome • web-javascript • Google Chrome 105.0.5195.125

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```

```bash
[✓] Flutter (Channel master, 3.4.0-28.0.pre.123, on macOS 12.6 21G115 darwin-x64, locale en-VN)
    • Flutter version 3.4.0-28.0.pre.123 on channel master at /Users/huynq/Documents/GitHub/flutter_master
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 4aa27d844e (3 minutes ago), 2022-09-22 21:00:22 -0700
    • Engine revision a5489ce4f9
    • Dart version 2.19.0 (build 2.19.0-229.0.dev)
    • DevTools version 2.17.0

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /Users/huynq/Library/Android/sdk
    • Platform android-33, build-tools 31.0.0
    • ANDROID_HOME = /Users/huynq/Library/Android/sdk
    • Java binary at: /Applications/Android Studio.app/Contents/jre/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)
    • All Android licenses accepted.

[✓] Xcode - develop for iOS and macOS (Xcode 14.0)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Build 14A309
    • CocoaPods version 1.11.3

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 2021.2)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 11.0.12+0-b1504.28-7817840)

[✓] IntelliJ IDEA Community Edition (version 2022.1.1)
    • IntelliJ at /Users/huynq/Library/Application Support/JetBrains/Toolbox/apps/IDEA-C/ch-0/221.5591.52/IntelliJ IDEA CE.app
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart

[✓] VS Code (version 1.71.2)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.48.0

[✓] Connected device (3 available)
    • SM T225 (mobile) • R9JT3004VRJ • android-arm64  • Android 12 (API 31)
    • macOS (desktop)  • macos       • darwin-x64     • macOS 12.6 21G115 darwin-x64
    • Chrome (web)     • chrome      • web-javascript • Google Chrome 105.0.5195.125
    ! Error: Failed to prepare the device for development. Please unlock and reconnect the device. (code 806)

[✓] HTTP Host Availability
    • All required HTTP hosts are available

• No issues found!

```
</details>

Please provide a [completed and minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) so that we may verify this. 
Thank you!

**Comment by matthewfx** [comment]
Hi, the same happens to me. However, in my case, I'm almost sure it is caused by the flavours I have. On iOS it launches the app and then gets stuck until after 12 minutes it displays a time-out error. On Android, it works fine. Even when I replace my app with a dummy app displaying a single text widget it also doesn't work. I'm guessing in my case it is the flavour param that is causing a problem. I don't have time to create another simple app and add flavours to it but I can't think of any other reasons atm

**Comment by AndreHaueisen** [comment]
Thanks @matthewfx  
This is probably it. I'm also using flavors.

**Comment by huycozy** [comment]
@AndreHaueisen @matthewfx So the issue won't happen when you run the tests with flavor, right?

**Comment by matthewfx** [comment]
@huycozy no it happens when you run with a flavor. If you try to run without a flavor Xcode doesn't build it and afaik there is no way to avoid it.

**Comment by AndreHaueisen** [comment]
TLDR: if your app use flavors, when you run integration tests on iOS, they get stuck

**Comment by huycozy** [comment]
@AndreHaueisen @matthewfx  thanks for the confirmation, could you please provide a [completed and minimal reproducible code sample](https://stackoverflow.com/help/minimal-reproducible-example) so that we may verify this? Also, please update the issue's content (title, desc, steps to reproduce) corresponding with this case as well (with a flavor).
Thank you!

**Comment by matthewfx** [comment]
My app is huge atm and I don't have time to create a sample app with flavours, sorry. @huycozy Have you tried to add flavours to the counter sample app?  I'm going to run integration tests on Android until you fix it.

**Comment by IgoByBobbyorPro** [comment]
Can the flutter integration take --target as an option for a proposed fix? For me, when I run the test it will open the app and just hang - assuming @matthewfx you are experiencing the same - even though within the `*.xcconfig` files there is an expected `FLUTTER_TARGET` already listed

**Comment by huycozy** [comment]
@matthewfx  I tried running the integration test sample from [integration_tests/flavors](https://github.com/flutter/flutter/blob/master/dev/integration_tests/flavors/integration_test/integration_test.dart) but it has done as expected without stuck.

Device info: iPhone 14, iOS 16.0

<details>
<summary>Demo</summary>


https://user-images.githubusercontent.com/104349824/194123513-d4c0d33b-9212-4e3b-916d-670466a2455d.mp4


</details>

Can you also try `integration_tests/flavors` sample code and confirm?

**Comment by matthewfx** [comment]
I'm very busy but will try to find some time. Thanks

**Comment by rddewan** [comment]
I have a same problem with flavor , it works fine with android but in iOS it get stuck

**Comment by matthewfx** [comment]
@rddewan Looks like we won't be able to test with iOS... They have closed it...

**Comment by rddewan** [comment]
@matthewfx it look like flavor not supported , without flavor it run

**Comment by huycozy** [comment]
Hi @matthewfx, @rddewan
I reopen this issue due to the bot closed it automatically. 
Can you try running [integration_tests/flavors](https://github.com/flutter/flutter/blob/master/dev/integration_tests/flavors/integration_test/integration_test.dart) to check if the test can run normally?

**Comment by rddewan** [comment]
@huycozy i am using [integration_test] package without flutter drive, do we still need to use flutter drive to make the integration test work with firebase test lab ?

**Comment by rddewan** [comment]
@huycozy  i cant run the (https://github.com/flutter/flutter/blob/master/dev/integration_tests/flavors/integration_test/integration_test.dart) so many dependency error with flutter 3.3.5
